win32ss/user/ntuser/window.c

```diff
@@ -3996,19 +3997,27 @@ NtUserQueryWindow(HWND hWnd, DWORD Index)
          Result = (pWnd->head.pti->MessageQueue == gpqForeground);
          break;
 
-      case QUERY_WINDOW_DEFAULT_IME:
-         ERR("QUERY_WINDOW_DEFAULT_IME: FIXME\n");
-         Result = 0;
+      case QUERY_WINDOW_DEFAULT_IME: /* default IME window */
+         Result = (DWORD_PTR)UserHMGetHandle(pWnd->head.pti->spwndDefaultIme);
```

Comment:
The `spwndDefaultIme` may not be initialized yet (and therefore can be NULL), and therefore you _**must**_ check whether it is NULL, before calling the `UserHMGetHandle` function. See what is done e.g. for the `QUERY_WINDOW_FOCUS` case above.

Reply:
Added NULL check.